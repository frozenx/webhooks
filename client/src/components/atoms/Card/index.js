import React from 'react';
import Label from '@beans/label';
import Button from '@beans/button';
import i18Data from '../../../languagepack';
import CardContainer from '../../templates/card';
import {
    CardTopContainer,
    CardBottomContainer,
    StyledLeftSection,
    StyledRightSection,
    StyledBodyText
} from './styled'
import StatusIndicator from '@beans/colleague-status-indicator';

const AddressBookCard = ({ address, navigateToForm }) => {
    const { address1, address2, address3 } = address;
    const delimiter = ',';
    let addressLine = address1;
    if (address2)
        addressLine += `${delimiter}${address2}`;
    if (address3)
        addressLine += `${delimiter}${address3}`;


    return (
        <CardContainer>
            <CardTopContainer>
                <StyledLeftSection>
                    <Label dark={true} emphasized={true}>{address1}</Label>
                </StyledLeftSection>
                <StyledRightSection>
                    <Button variant="secondary" onClick={() => { navigateToForm(address.addressUuid) }}>
                        {i18Data.view}
                    </Button>
                </StyledRightSection>
            </CardTopContainer>
            <CardBottomContainer>
                <StyledLeftSection>
                    <Label dark={true} emphasized={true}>{i18Data.address}</Label>
                    {address1 && <StyledBodyText>{`${addressLine}`}</StyledBodyText>}
                </StyledLeftSection>
                <StyledRightSection>
                    <Label dark={true} emphasized={true}>{i18Data.status}</Label>
                    <Label>
                        {<StatusIndicator variant={address.isActive === 'Active' ? "success" : "error"}>
                            {address.isActive || "I"}
                        </StatusIndicator>
                        }
                    </Label>
                </StyledRightSection>
            </CardBottomContainer>
        </CardContainer>
    )
}
export default AddressBookCard;
