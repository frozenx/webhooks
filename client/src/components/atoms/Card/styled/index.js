import styled from 'styled-components'
import { media } from '@beans/foundation';
import { BodyText } from '@beans/typography';


export const CardTopContainer = styled.div`
    label {
        font-size: 24px;
    }
    display: flex;
    ${media.belowTablet`flex-wrap: wrap;`};
`

export const CardBottomContainer = styled.div` 

    width: 100%;
    display: flex;
    margin-top: 8px;
    ${media.belowTablet`flex-wrap: wrap;`};
`
export const StyledLeftSection = styled.div`
    display: inline-block;
    margin-right: 10px;
    margin-bottom: ${props => props.spaceBottom && '20px' };
    label{
        width: 100%;  
    }
    margin-top: 8px;
    min-width: 434px;
    text-wrap: wrap;
    
`
export const StyledRightSection = styled(StyledLeftSection)`
    margin-right: 0px;
    margin-top: 8px;
`
export const StyledBodyText = styled(BodyText)`
    text-wrap: wrap;
`;
