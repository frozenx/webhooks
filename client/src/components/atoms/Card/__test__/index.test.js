import React from 'react';
import Card from '../'
import {shallow} from 'enzyme'
import renderer from 'react-test-renderer';

describe('atoms/Card', () => {
    describe('Given the card component is mounted', () => {
        test('it should render the markup' , () => {
            const tree = renderer.create(<Card/>).toJSON();
            expect(tree).toMatchSnapshot()
        })
        test('it should trigger the onclick for view button' , () => {
            const mockProps = {
                handleClick: jest.fn()
            }
            const tree = shallow(<Card {...mockProps}/>)
            const viewButton = tree.find('_default').at(1);
            viewButton.simulate('click');
            //expect(mockProps.handleClick).toHaveBeenCalled();
        })
    })
})