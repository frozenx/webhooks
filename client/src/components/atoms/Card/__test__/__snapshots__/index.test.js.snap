// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms/Card Given the card component is mounted it should render the markup 1`] = `
<styleGrayWrapper>
  <cardTopContainer />
  <cardBottomContainer />
</styleGrayWrapper>
`;
